play-json-ops-common-213/src/main/scala/play/api/libs/json/ops/v4/DurationFormat.scala

```diff
@@ -0,0 +1,159 @@
+package play.api.libs.json.ops.v4
+
+import play.api.libs.json._
+
+import scala.concurrent.duration._
+import scala.concurrent.duration.ops.v4.DurationOps
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
+
+/**
+ * Provides implicit readers for Scala [[Duration]]s.
+ */
+trait ImplicitDurationReads {
+
+  implicit protected def finiteDurationReads: Reads[FiniteDuration]
+
+  implicit protected def durationReads: Reads[Duration]
+}
```

Comment:
This is a terrible pattern for binary compatibility, but I think we have a lot of things to clean up if we want to make any changes here. I would suggest we make forward binary compatibility updates in v5. Although, maybe this code could serve as an example... so I might tackle making this change before merging.